File: src/test/java/co/nlighten/jsontransform/functions/TransformerFunctionCsvTest.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.BaseTest;
import org.junit.jupiter.api.Test;
public class TransformerFunctionCsvTest extends BaseTest {
    @Test
    void inline() {
        assertTransformation(fromJson("""
               [{"a":"A","b":1},{"a":"C","b":2}]"""),
             "$$csv:$",
             """
a,b
A,1
C,2
""");
        assertTransformation(fromJson("""
               [{"a":"A","b":1},{"a":"C","b":2}]"""),
             "$$csv(true):$",
             """
A,1
C,2
""");
    }

    @Test
    void object() {
        assertTransformation(
                fromJson("""
               [{"a":"A","b":1},{"a":"C","b":2}]"""),
                fromJson("""
                    {
                      "$$csv": "$"
                    }"""), """
a,b
A,1
C,2
""");
        assertTransformation(
                fromJson("""
               [{"a":"A","b":1},{"a":"C","b":2}]"""),
                fromJson("""
                    {
                      "$$csv": "$",
                      "no_headers": true
                    }"""), """
A,1
C,2
""");
   }

    @Test
    void testObject_names() {
        assertTransformation(
               fromJson("[[1,2],[3,4]]"),
               fromJson("""
                    {
                      "$$csv": "$",
                      "names": ["a","b"]
                    }"""), """
a,b
1,2
3,4
""");
        // without names
        assertTransformation(
                fromJson("[[1,2],[3,4]]"),
                fromJson("""
                    {
                      "$$csv": "$"
                    }"""), """
1,2
3,4
""");
    }
}

File: javascript/json-transform/src/__tests__/functions/TransformerFunctionCsv.test.ts
import { describe, test } from "vitest";
import { assertTransformation } from "../BaseTransformationTest";

describe("TransformerFunctionCsv", () => {
  test("inline", async () => {
    await assertTransformation(
      [
        { a: "A", b: 1 },
        { a: "C", b: 2 },
      ],
      "$$csv:$",
      "a,b\nA,1\nC,2\n",
    );
    await assertTransformation(
      [
        { a: "A", b: 1 },
        { a: "C", b: 2 },
      ],
      "$$csv(true):$",
      "A,1\nC,2\n",
    );
  });

  test("object", async () => {
    await assertTransformation(
      [
        { a: "A", b: 1 },
        { a: "C", b: 2 },
      ],
      {
        $$csv: "$",
      },
      "a,b\nA,1\nC,2\n",
    );
    await assertTransformation(
      [
        { a: "A", b: 1 },
        { a: "C", b: 2 },
      ],
      {
        $$csv: "$",
        no_headers: true,
      },
      "A,1\nC,2\n",
    );
  });

  test("object_names", async () => {
    await assertTransformation(
      [
        [1, 2],
        [3, 4],
      ],
      {
        $$csv: "$",
        names: ["a", "b"],
      },
      "a,b\n1,2\n3,4\n",
    );
    // without names
    await assertTransformation(
      [
        [1, 2],
        [3, 4],
      ],
      { $$csv: "$" },
      "1,2\n3,4\n",
    );
  });
});